tests/jobserver/authorization/test_fields.py

```diff
@@ -1,12 +1,27 @@
 import pytest
 from django.db import connection
 
-from jobserver.authorization import CoreDeveloper
-from jobserver.authorization.fields import RolesField
+from jobserver.authorization import CoreDeveloper, OutputChecker
+from jobserver.authorization.fields import ExtractRoles, RolesField
+from jobserver.models import User
 
 from ...factories import OrgFactory, OrgMembershipFactory, UserFactory
 
 
+@pytest.mark.django_db
+def test_extractroles():
+    user1 = UserFactory(roles=[OutputChecker])
+    user2 = UserFactory(roles=[CoreDeveloper])
+
+    users = User.objects.alias(extracted=ExtractRoles("roles")).filter(
+        extracted__contains="CoreDeveloper"
```

Comment:
Does this want to be `'"CoreDeveloper"` i.e to match on the quotes as well? Otherwise it would match the role `NotACoreDeveloper`, which is obviously a silly example but you can imagine other cases where that might happen and it would be good to establish the right pattern here.

Reply:
The full string for the role is the dotted path to the defined class, so in this case `jobserver.authorization.roles.CoreDeveloper`, an implementation detail I wanted to hide from a consumer of the roles APIs, but you're right that we could potentially add roles which cause problems, we already have three Project* roles already!

A better fit with the current use of Roles would be passing the class and letting the framework handle the conversion, so really this should be abstracted off to a QuerySet/Manager on the Model.